modules/c++/net.ssl/wscript

```diff
@@ -2,8 +2,119 @@ NAME            = 'net.ssl'
+
+    class NetSSLCallback:
+        def __init__(self):
+            self.uselib = None
+            self.includes = None
+            self.libpath = None
+
+        def __call__(self, conf):
+            sslHome = Options.options.ssl_home
+            sslLib = 'ssl'
+            sslUselib = 'SSL'
+            sslHeader = 'openssl/ssl.h'
+            sslFunction = 'SSL_library_init'
+            sslDef = 'USE_OPENSSL'
+            sslOldDef = 'OPENSSL_0_9_8'
+            sslFragment098 = '''
+                #include "openssl/ssl.h"
+                int main()
+                {
+                    SSL_library_init();
+                    SSL_load_error_strings();
+
+                    SSL_METHOD *method = SSLv23_client_method();
+                    SSL_CTX* ctx = SSL_CTX_new(method);
+                    return 0;
+                }
+                '''
+            sslFragment100 = '''
+                #include "openssl/ssl.h"
+                int main()
+                {
+                    SSL_library_init();
+                    SSL_load_error_strings();
+
+                    const SSL_METHOD *method = SSLv23_client_method();
+                    SSL_CTX* ctx = SSL_CTX_new(method);
+                    return 0;
+                }
+                '''
+
+            if sslHome:
+                if conf.check(lib=sslLib, uselib_store=sslUselib,
+                              header_name=sslHeader,
+                              function_name=sslFunction,
+                              fragment=sslFragment098,
+                              libpath=os.path.join(sslHome, 'lib'),
+                              includes=os.path.join(sslHome, 'include'),
+                              msg='Checking for version 0.9.8 of ssl library',
+                              okmsg=sslHome,
+                              define_name=sslOldDef,
+                              mandatory=False):
+                    conf.define(sslOldDef, True)
+                else:
+                    conf.check(lib=sslLib, uselib_store=sslUselib,
+                               header_name=sslHeader,
+                               function_name=sslFunction,
+                               fragment=sslFragment100,
+                               libpath=os.path.join(sslHome, 'lib'),
+                               includes=os.path.join(sslHome, 'include'),
+                               msg='Checking for version 1.0.0 of ssl library',
+                               okmsg=sslHome,
+                               mandatory=True)
+                conf.define(sslDef, True)
+            elif conf.check(lib=sslLib, uselib_store=sslUselib,
+                            header_name=sslHeader,
+                            function_name=sslFunction,
+                            fragment=sslFragment098,
+                            msg='Checking for version 0.9.8 of ssl library',
+                            mandatory=False):
+                conf.define(sslDef, True)
```

Comment:
It does work with both of them and this way there are no #defines an external user would have to set at all. The downside is that the order of the #includes matter now within the net.ssl module. Would you rather have have USE_OPENSSL be a conf define?